File: player/src/main/java/com/libreshockwave/player/PlayerState.java
package com.libreshockwave.player;

/**
 * Playback state of the Director player.
 */
public enum PlayerState {
    /** Player is stopped, no movie loaded */
    STOPPED,
    /** Player is paused, movie loaded but not advancing */
    PAUSED,
    /** Player is playing, advancing frames */
    PLAYING
}

File: cpp/src/player/PlayerState.cpp
#include "libreshockwave/player/PlayerState.hpp"

#include <stdexcept>

namespace libreshockwave::player {

std::string_view name(PlayerState state) {
    switch (state) {
        case PlayerState::Stopped: return "STOPPED";
        case PlayerState::Paused: return "PAUSED";
        case PlayerState::Playing: return "PLAYING";
    }
    throw std::logic_error("Unknown PlayerState enum value");
}

} // namespace libreshockwave::player
